src/kernel.ts

```diff
@@ -884,12 +905,15 @@ class Kernel implements IKernel {
       return;
     }
     var content = msg.content as ICommOpen;
-    if (!content.target_module) {
-      this.commOpened.emit(msg);
+    this.commOpened.emit(msg);
+    var targetName = content.target_name;
+    var moduleName = content.target_module;
+    if (this._targetRegistry[targetName]) {
+      this._targetRegistry[targetName](msg);
+    }
```

Comment:
I see real a benefit using to a registry based approach, at least in ipywidgets, because it allows the user to hack on things in the live notebook.  It's also guaranteed to work as long as we support it, where methods to define arbitrary modules, like `define('mymodule', function() {})` provided by require.js, are not guaranteed to exist and depend on the loader we use in the frontend.